server/src/main/java/org/eclipse/lsp/cobol/service/CopybookNameServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.service;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.google.inject.Singleton;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.lsp.cobol.jrpc.CobolLanguageClient;
+import org.eclipse.lsp.cobol.service.utils.FileSystemService;
+import org.eclipse.lsp4j.WorkspaceFolder;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.unmodifiableList;
+import static org.eclipse.lsp.cobol.service.utils.SettingsParametersEnum.CPY_LOCAL_PATHS;
+
+/**
+ * This service processes all the copybook names present in the local directory. The service also
+ * caches copybook to reduce filesystem load.
+ */
+@Slf4j
+@Singleton
+public class CopybookNameServiceImpl implements CopybookNameService {
+  private static List<String> resolvedCopybookNames;
+  private static List<String> listofCopyBooks = new ArrayList<>();
+  private final FileSystemService files;
+  private final Provider<CobolLanguageClient> clientProvider;
+  private final SettingsService settingsService;
+
+  @Inject
+  public CopybookNameServiceImpl(
+      SettingsService settingsService,
+      FileSystemService files,
+      Provider<CobolLanguageClient> clientProvider) {
+    this.settingsService = settingsService;
+    this.files = files;
+    this.clientProvider = clientProvider;
+    this.resolvedCopybookNames = new ArrayList<>();
+  }
+
+  @Override
+  public Collection<String> getNames() {
+    return unmodifiableList(resolvedCopybookNames);
+  }
+
+  @Override
+  public void collectLocalCopybookNames() {
+    listofCopyBooks.clear();
+    clientProvider
+        .get()
+        .workspaceFolders()
+        .thenAcceptBoth(
+            settingsService.getConfiguration(CPY_LOCAL_PATHS.label), this::resolveNames);
+  }
+
+  @Override
+  public void store(List<String> name) {
+    resolvedCopybookNames.addAll(name);
+  }
+
+  private void resolveNames(List<WorkspaceFolder> workspace, List<Object> copybookLocal) {
+    settingsService
+        .toStrings(copybookLocal)
+        .forEach(
+            fileName ->
+                workspace.stream()
+                    .map(
+                        path ->
+                            files
+                                .getPathFromURI(files.decodeURI(path.getUri()))
```

Comment:
Resolving the paths against the workspace folders means that the completion of the external copybooks won't work.